core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -242,7 +243,11 @@ class LogManagerTest {
   def testRecoveryDirectoryMappingWithRelativeDirectory() {
     logManager.shutdown()
     logDir = new File("data" + File.separator + logDir.getName)
-    logDir.mkdirs()
+    try {
+          Files.createDirectory(logDir.toPath())    
+      } catch {
+        case e: IOException => //this is good
```

Comment:
I wanted to encompass the createDirectory call in a try catch block but didn't want the test to be stop when an exception was thrown so I just commented that after not handling the exception once caught. What would be a better way to do this?